### PR TITLE
[ISSUE #172][ISSUE #174]Modify the File Source Example; Update README; Fix incompatible types bug

### PR DESCRIPTION
**Pull request by hyl-xidian**
Add a filter usage example and modify the document.

### CODE REVIEW

**Comment by hyl-xidian** [comment]
#174 fix incompatible types bug

File: rocketmq-streams-commons/src/main/java/org/apache/rocketmq/streams/common/context/UserDefinedMessage.java [hunk 1]
```diff
@@ -318,12 +318,12 @@ public Date getDate(String key) {
 
     @Override
     public java.sql.Date getSqlDate(String key) {
-        return super.getSqlDate(key);
+        return (java.sql.Date) super.getSqlDate(key);
```
Comment (ni-ze) [inline rocketmq-streams-commons/src/main/java/org/apache/rocketmq/streams/common/context/UserDefinedMessage.java:321]: It is cause by update fastjson version, and I fixed it by changed the return type.

**Comment by ni-ze** [comment]
https://github.com/apache/rocketmq-streams/issues/172

**Comment by ni-ze** [comment]
@hyl-xidian can you remove the fix bug part? The others looks fine to me.

**Comment by hyl-xidian** [comment]
> @hyl-xidian can you remove the fix bug part? The others looks fine to me.

Got it, I have removed the fix bug part.

**Pull request merged**